### PR TITLE
Vertical scaling - Scheduler, CommonInfo, and unit-test changes

### LINKED ISSUE #223: Vertical Scaling for Arktos
**What would you like to be added**: Vertical Scaling Feature for Arktos

**Why is this needed**: This feature supports optimal resource management and better usage of cloud resources.

The following tasks have been identified in delivery of this feature.

- [ ] Design Document for Container Pods Vertical Scaling
- [ ] Design Document for Virtual Machine Pods Vertical Scaling
- [ ] API Changes for Containers
- [ ] API Changes to support Virtual Machine
- [ ] Scheduler Changes
- [ ] Kubelet Changes for Container Pods
- [ ] CRI changes for Container Pods
- [ ] Kubelet Changes for VM Pods
- [ ] CRI/Virtlet Changes to support VMs
- [ ] E2E Tests for Containers
- [ ] E2E Tests for VMs

### PR DESCRIPTION
**Pull request by vinaykul**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This extends Vertical Scaling API support for Arktos VM / Container Pods, and includes Scheduler changes to support Vertical Scaling.

**Which issue(s) this PR fixes**:

Fixes #223 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/aws/util.sh [hunk 1]
```diff
@@ -920,6 +920,11 @@ function start-flannel-ds {
   popd
 }
 
+function start-weave-ds {
```
Comment (XiaoningDing) [inline cluster/aws/util.sh:923]: Are these two cluster/aws files irrelevant to this PR?

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -3745,15 +3744,6 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("activeDeadlineSeconds"), newPod.Spec.ActiveDeadlineSeconds, "must not update from a positive integer to nil value"))
 	}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
-		// reject QOS change attempt
```
Comment (XiaoningDing) [inline pkg/apis/core/validation/validation.go:3749]: Could you explain why this check is removed. Anything to do with VM?

File: plugin/pkg/admission/podresourceallocation/admission.go [hunk 3]
```diff
@@ -149,7 +149,10 @@ func (p *Plugin) Validate(attributes admission.Attributes, o admission.ObjectInt
 	case admission.Create:
 		// newly created pods must have ResourcesAllocated field either empty or equal to Requests
 		for _, c := range pod.Spec.Containers {
-			if c.ResourcesAllocated == nil ||
+			if c.Resources.Requests == nil {
```
Comment (XiaoningDing) [inline plugin/pkg/admission/podresourceallocation/admission.go:152]: Will this admit work for guaranteed pods, which have only resources.limit set?  Or this admission is after an admission that will change resources.request for guaranteed pods?

File: cluster/aws/util.sh [hunk 4]
```diff
@@ -920,6 +920,11 @@ function start-flannel-ds {
...
+  local kubectl_ver="$(${KUBE_ROOT}/cluster/kubectl.sh version | base64 | tr -d '\n')"
```
Comment (XiaoningDing) [inline cluster/aws/util.sh:924]: why is weave network and pod-network-cidr stuff related to your changes? If it's for something else please move them to a separate PR. thanks.

Comment (vinaykul) [reply cluster/aws/util.sh:924]: Moved to PR https://github.com/futurewei-cloud/arktos/pull/235
